### PR TITLE
[vcpkg baseline][paraview] Fix linux build

### PR DESCRIPTION
**Pull request by JackBoosY**
- Add feature `tools`
- Add dependency `embree3` to paraview on Linux.
- Disable feature `tools` on OSX.

Other changes use @Neumann-A suggestion https://github.com/microsoft/vcpkg/pull/11596#issuecomment-650833171.

Related: #12018 #12126 #11754 #12145 #12068 etc.

### CODE REVIEW

File: ports/paraview/portfile.cmake [hunk 1]
```diff
@@ -6,6 +6,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     "mpi"          PARAVIEW_USE_MPI             #untested
     "vtkm"         PARAVIEW_USE_VTKM
     "python"       PARAVIEW_USE_PYTHON
+    "tools"        PARAVIEW_BUILD_TOOLS
 )
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:11]:
```suggestion

if ("tools" IN_LIST FEATURES)
    message(FATAL_ERROR "Feature tools is not available now,  waiting for fix. ")
endif()
    

```

File: ports/paraview/portfile.cmake [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+    mesage(FATAL_ERROR "Feature tools are currently not supported, waiting for fix it.")
+endif()
+
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     "cuda"         PARAVIEW_USE_CUDA            #untested; probably only affects internal VTK build so it does nothing here
     "all_modules"  PARAVIEW_BUILD_ALL_MODULES   #untested
     "mpi"          PARAVIEW_USE_MPI             #untested
-    "vtkm"         PARAVIEW_USE_VTKM
+
+if ("tools" IN_LIST FEATURES)
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:12]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+    message(FATAL_ERROR "Feature tools is not available now,  waiting for fix. ")
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:13]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 4]
```diff
@@ -1,11 +1,21 @@
...
+endif()
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:14]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+if ("tools" IN_LIST FEATURES)
+    mesage(FATAL_ERROR "Feature tools are currently not supported, waiting for fix it.")
+endif()
+
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     "cuda"         PARAVIEW_USE_CUDA            #untested; probably only affects internal VTK build so it does nothing here
     "all_modules"  PARAVIEW_BUILD_ALL_MODULES   #untested
     "mpi"          PARAVIEW_USE_MPI             #untested
-    "vtkm"         PARAVIEW_USE_VTKM
+
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:11]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 6]
```diff
@@ -1,11 +1,21 @@
...
+if ("tools" IN_LIST FEATURES)
+    message(FATAL_ERROR "Feature tools is not available now,  waiting for fix. ")
+endif()
+
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:15]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 7]
```diff
@@ -1,11 +1,21 @@
...
     "cuda"         PARAVIEW_USE_CUDA            #untested; probably only affects internal VTK build so it does nothing here
     "all_modules"  PARAVIEW_BUILD_ALL_MODULES   #untested
     "mpi"          PARAVIEW_USE_MPI             #untested
-    "vtkm"         PARAVIEW_USE_VTKM
+
+if ("tools" IN_LIST FEATURES)
+    message(FATAL_ERROR "Feature tools is not available now,  waiting for fix. ")
+endif()
+    
+
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:16]:
```suggestion
```

File: ports/paraview/portfile.cmake [hunk 8]
```diff
@@ -1,11 +1,16 @@
...
+if ("tools" IN_LIST FEATURES)
+    mesage(FATAL_ERROR "Feature tools are currently not supported, waiting for fix it.")
+endif()
+
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     "cuda"         PARAVIEW_USE_CUDA            #untested; probably only affects internal VTK build so it does nothing here
     "all_modules"  PARAVIEW_BUILD_ALL_MODULES   #untested
     "mpi"          PARAVIEW_USE_MPI             #untested
-    "vtkm"         PARAVIEW_USE_VTKM
+
```
Comment (JackBoosY) [inline ports/paraview/portfile.cmake:11]:
```suggestion
    "vtkm"         PARAVIEW_USE_VTKM
```

**Comment by JackBoosY** [comment]
Modify by my iPhone,  OMG.

**Comment by Neumann-A** [comment]
@JackBoosY: Updating vtk to 9.0.1 also solves the hdf5 issue. see #12149. The libmath issues still exists but it is probably not due to VTK but some other dependent library.

**Comment by Neumann-A** [comment]
@JackBoosY:  [VCPKG-DEBUG]: Found libmath.a in embree3

File: ports/paraview/portfile.cmake [hunk 9]
```diff
@@ -66,6 +72,9 @@ if("python" IN_LIST FEATURES)
     #VTK_PYTHON_SITE_PACKAGES_SUFFIX should be set to the install dir of the site-packages
 endif()
 
+string(APPEND VCPKG_C_FLAGS " -DH5_USE_110_API")
```
Comment (BillyONeal) [inline ports/paraview/portfile.cmake:75]: The `fix-using-hdf5.patch` should be removed above.

**Comment by Neumann-A** [comment]
Filled:
https://gitlab.kitware.com/paraview/paraview/-/issues/20043
PR can be closed since the temporary fix without patches is to add a build-depends on embree3

**Comment by BillyONeal** [comment]
@Neumann-A Are you saying we should not merge this? I think it makes sense to merge it even if only to get the better hdf5 fix.

**Pull request merged**